### PR TITLE
Denyactionpr

### PR DESCRIPTION
**Pull request by JamJarchitect**
## Overview/Summary

Implementation of DenyAction Delete into Azure Landing Zones for both Activity Logs and Diagnostic Settings as two seperate policy definitions and a policy initiative. 

Address story [AB#29619](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/29619)

## This PR fixes/adds/changes/removes

1. Two policy definitions for DenyAction Delete Activity Logs and Diagnostic Settings
2. A policy initiative to apply these two policy definitions

### Breaking Changes

1. N/A

## Testing Evidence

image

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fus29619%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fus29619%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fus29619%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fus29619%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: docs/wiki/Whats-new.md [hunk 1]
```diff
@@ -49,6 +49,10 @@ Major update in this release: introducing the Policy Testing Framework foundatio
   - Deploy-SQL-minTLS
   - Deploy-MySQL-sslEnforcement (changed from Owner to Contributor role, no built in roles currently available)
   - Deploy-PostgreSQL-sslEnforcement (changed from Owner to Contributor role, no built in roles currently available)
+  - Additional policy definitions:
```
Comment (Springstone) [inline docs/wiki/Whats-new.md:52]:
```suggestion
  - Adding `DenyAction` policy definitions:
```

File: docs/wiki/Whats-new.md [hunk 2]
```diff
@@ -49,6 +49,10 @@ Major update in this release: introducing the Policy Testing Framework foundatio
...
+    - `DenyAction-ActivityLogs`
+    - `DenyAction-DiagnosticLogs`
+    - These two policy definitions prevent Activity Log Settings and Diagnostic Settings being deleted leveraging Azure Policy DenyAction functionality
```
Comment (Springstone) [inline docs/wiki/Whats-new.md:55]:
```suggestion
    - These two policy definitions prevent Activity Log Settings and Diagnostic Settings being deleted leveraging Azure Policy DenyAction functionality
    - Additionally including an initiative that includes both `DenyAction` policies
```

**Review by Springstone — CHANGES_REQUESTED** [review]
Minor What's New updates

**Comment by jtracey93** [comment]
@JamJarchitect & @Springstone are we also looking to add tests for this one?

**Comment by jtracey93** [comment]
@Springstone @JamJarchitect - whats the latest with this one?

**Pull request merged**